src/components/Header/Header.tsx

```diff
@@ -0,0 +1,125 @@
+import React, { useEffect, useRef, useState } from 'react';
+import cn from 'classnames';
+import { USER_ID, addTodos, updateTodo } from '../../api/todos';
+import { ErrorMessages } from '../../types/ErrorMessages';
+import { Todo } from '../../types/Todo';
+
+interface Props {
+  todos: Todo[];
+  setTodos: (todos: Todo[]) => void;
+  setTempTodo: (tempTodo: Todo | null) => void;
+  setErrorMessage: (message: string) => void;
+  loadingIds: number[];
+  setLoadingIds: React.Dispatch<React.SetStateAction<number[]>>;
+}
+
+export const Header: React.FC<Props> = ({
+  todos,
+  setTodos,
+  setTempTodo,
+  setErrorMessage,
+  loadingIds,
+  setLoadingIds,
+}) => {
+  const inputRef = useRef<HTMLInputElement>(null);
+  const [title, setTitle] = useState('');
+  const [loading, setLoading] = useState(false);
+
+  const allCompleted = todos.every(todo => todo.completed);
+
+  useEffect(() => {
+    inputRef.current?.focus();
+  }, [todos, loading]);
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    const trimmedTitle = title.trim();
+
+    if (!trimmedTitle) {
+      setErrorMessage(ErrorMessages.EmptyTitle);
+
+      return;
+    }
+
+    setLoading(true);
+
+    const newTodo: Todo = {
+      id: 0,
+      userId: USER_ID,
+      title: trimmedTitle,
+      completed: false,
+    };
+
+    setTempTodo({ ...newTodo, id: 0 });
+
+    addTodos(newTodo)
+      .then((todo: Todo) => {
+        setTodos([...todos, todo]);
+        setTitle('');
+      })
+      .catch(() => {
+        setErrorMessage(ErrorMessages.UnableAdd);
+        setTitle(newTodo.title);
+      })
+      .finally(() => {
+        setLoading(false);
+        setTempTodo(null);
+      });
+  };
+
+  const completedForAll = () => {
+    const updatedTodos = todos.map(todo => ({
+      ...todo,
+      completed: !allCompleted,
+    }));
+
+    const updatePromises = todos.map(todo =>
+      todo.completed === allCompleted
+        ? updateTodo({ ...todo, completed: !allCompleted })
+        : Promise.resolve(),
+    );
+
+    const todoIds = updatedTodos.map(todo => todo.id);
+
+    setLoadingIds(todoIds);
+
+    Promise.all(updatePromises)
+      .then(() => {
+        setTodos(updatedTodos);
```

Comment:
The same. And one more problem here, it toggles all even some of them don't need the update